src/transformers/modeling_tf_bert.py

```diff
@@ -919,78 +1144,106 @@ def call(
             output_attentions=output_attentions,
             output_hidden_states=output_hidden_states,
             return_dict=return_dict,
+            labels=labels,
             training=training,
         )
-
+        return_dict = inputs["return_dict"] if inputs["return_dict"] is not None else self.bert.return_dict
+        outputs = self.bert(
+            input_ids=inputs["input_ids"],
+            attention_mask=inputs["attention_mask"],
+            token_type_ids=inputs["token_type_ids"],
+            position_ids=inputs["position_ids"],
+            head_mask=inputs["head_mask"],
+            inputs_embeds=inputs["inputs_embeds"],
+            output_attentions=inputs["output_attentions"],
+            output_hidden_states=inputs["output_hidden_states"],
+            return_dict=return_dict,
+            training=inputs["training"],
+        )
         sequence_output = outputs[0]
-        prediction_scores = self.mlm(sequence_output, training=training)
-        loss = None if labels is None else self.compute_loss(labels, prediction_scores)
+        prediction_scores = self.cls(sequence_output=sequence_output, training=training)
+        lm_loss = None
 
-        if not return_dict:
-            output = (prediction_scores,) + outputs[2:]
-            return ((loss,) + output) if loss is not None else output
+        if inputs["labels"] is not None:
+            # shift labels to the left and cut last logit token
+            shifted_prediction_scores = prediction_scores[:, :-1]
+            labels = inputs["labels"][:, 1:]
+            lm_loss = self.compute_loss(labels=labels, logits=shifted_prediction_scores)
 
-        return TFMaskedLMOutput(
-            loss=loss,
+        if tf.executing_eagerly():
+            if not return_dict:
+                output = (prediction_scores,) + outputs[2:]
+
+                return ((prediction_scores,) + output) if lm_loss is not None else output
```

Comment:
should this be returning the `lm_loss`?
```suggestion
                return ((lm_loss,) + output) if lm_loss is not None else output
```